public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/AlertManagerRouting.tsx

```diff
@@ -0,0 +1,122 @@
+import { css } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Alert, CollapsableSection, Icon, Link, LoadingPlaceholder, Stack, Text, useStyles2 } from '@grafana/ui';
+import { AlertManagerDataSource } from 'app/features/alerting/unified/utils/datasource';
+import { createUrl } from 'app/features/alerting/unified/utils/url';
+
+import { useContactPointsWithStatus } from '../../../contact-points/useContactPoints';
+
+import { ContactPointDetails } from './contactPoint/ContactPointDetails';
+import { ContactPointSelector } from './contactPoint/ContactPointSelector';
+import { MuteTimingFields } from './route-settings/MuteTimingFields';
+import { RoutingSettings } from './route-settings/RouteSettings';
+
+interface AlertManagerManualRoutingProps {
+  alertManagerContactPoint: {
+    alertManager: AlertManagerDataSource;
+    selectedContactPoint: string;
+    muteTimeIntervals: string[];
+    overrideGrouping: boolean;
+    groupBy: string[];
+    overrideTimings: boolean;
+    groupWaitValue: string;
+    groupIntervalValue: string;
+    repeatIntervalValue: string;
+  };
+}
+
+export function AlertManagerManualRouting({ alertManagerContactPoint }: AlertManagerManualRoutingProps) {
+  const styles = useStyles2(getStyles);
+
+  const alertManagerName = alertManagerContactPoint.alertManager.name;
+  const { isLoading, error: errorInContactPointStatus, contactPoints } = useContactPointsWithStatus();
+  const selectedContactPoint = contactPoints.find((cp) => cp.name === alertManagerContactPoint.selectedContactPoint);
```

Comment:
Is there a way to pass the entire `ContactPoint` down to this component from the drop-down selection so we don't have to fetch and find again?

Reply:
We are only fetching once. But we can keep the state of the select contact point with the metadata here to avoid this `find`.